### PR TITLE
BFD-3660: Initial SAMHSA V2 RFC

### PR DESCRIPTION
**Pull request by aschey-forpeople**
**JIRA Ticket:**
BFD-3660

### What Does This PR Do?

Adds initial RFC draft for the SAMHSA V2 changes.

[rendered](https://github.com/CMSgov/beneficiary-fhir-data/blob/BFD-3660/docs/rfcs/0023-samhsa-v2.md)

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:  

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data
* Any other changes that could possibly affect security? 

* [x] I have considered the above security implications as it relates to this PR. (If one or more of the above apply, it cannot be merged without the ISSO or team security engineer's (`@sb-benohe`) approval.) 

### Validation

Have you fully verified and tested these changes? Is the acceptance criteria met? Please provide reproducible testing instructions, code snippets, or screenshots as applicable.

### CODE REVIEW

**Review by dondevun — DISMISSED** [review]
This looks great; seems like a sound, high level  description of the solution.

**Review by malessi — DISMISSED** [review]
This RFC is a great summary of the proposal. _So_ much simpler to understand than the current filter 😃

File: docs/rfcs/0023-samhsa-v2.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+The current filter logic is tightly coupled to the implementation of the FHIR transformers.
+Any change to the system URLs, additional functionality that adds a potentially sensitive code to a new field in the FHIR response, or a bug that causes certain fields to not be mapped in the FHIR payload may cause the filter to behave incorrectly. 
+
+Instead of basing the SAMHSA filter on the FHIR payload, it would be safer to check any database column that could contain sensitive info and explicitly mark that claim as sensitive at ingestion time.
+
+## Implementation Overview
+
+The implementation will utilize [FHIR Security Labels](https://build.fhir.org/security-labels.html) to tag claims as 'sensitive'.
+SAMHSA claims will include a [Confidentiality](https://terminology.hl7.org/4.0.0/CodeSystem-v3-Confidentiality.html) tag of `R` for 'restricted' and an [Act Code](https://terminology.hl7.org/6.0.2/CodeSystem-v3-ActCode.html) tag of `42CFRPart2`.
```
Comment (mjburling) [inline docs/rfcs/0023-samhsa-v2.md:43]: I think the content is nearly or entirely identical, but it's important that we clearly refer to the like-sequenced FHIR™ R4 and/or 4.0.1 where possible.
```suggestion
The implementation will utilize [FHIR Security Labels](https://hl7.org/fhir/R4/security-labels.html) to tag claims as 'sensitive'.
SAMHSA claims will include a [Confidentiality](https://terminology.hl7.org/5.5.0/CodeSystem-v3-Confidentiality.html#v3-Confidentiality-R) tag of `R` for 'restricted' and an [Act Code](https://terminology.hl7.org/5.5.0/CodeSystem-v3-ActCode.html#v3-ActCode-42CFRPart2CD) tag of `42CFRPart2`.
```

File: docs/rfcs/0023-samhsa-v2.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+## Pipeline Updates
+
+The CCW and RDA pipelines will need to be updated to populate the new tables.
+At processing time, the pipeline will check each claim against the configured list of SAMHSA codes and populate the tag tables with both the `R` and `42CFRPart2` tags for each SAMHSA claim.
+
+## FHIR Server Updates
+
+The SQL queries used to fetch the claims data for the EOB, Claim, and ClaimResponse endpoints will need to join against the tag tables in order to determine which claims have SAMHSA data.
+SAMHSA claims should not be included if `excludeSAMHSA=true` is supplied in the request. 
+The `R` and `42CFRPart2` [Security Labels](https://build.fhir.org/security-labels.html) should be applied to the FHIR response if SAMHSA data is allowed to be returned.
```
Comment (mjburling) [inline docs/rfcs/0023-samhsa-v2.md:214]:
```suggestion
The `R` and `42CFRPart2` [Security Labels](https://hl7.org/fhir/R4/security-labels.html) should be applied to the FHIR response if SAMHSA data is allowed to be returned.
```

**Comment by alex-dzeda** [comment]
> I made some minor comments on the specific references to FHIR™ R4 and so, but with the rest of the team, I'm overall pleased with this direction!
> 
> I am a little curious about this implementation in the context of STU3/BFD v1, though. Maybe @alex-dzeda can provide some additional guidance here. While [STU3 security labels](https://hl7.org/fhir/STU3/security-labels.html) are supported, I see that HL7 Terminology was first published for R4/4.0.1. Does any of that matter for the content of the security labels?

That's fine, it just switched URLs between the release of the STU3 spec and R4, but the underlying OIDs are the same, so the content is, as well. At this point, to resolve the system, using the R4 style would be necessary.

File: docs/rfcs/0023-samhsa-v2.md (hunk 1 above)
Comment (aschey-forpeople) [reply docs/rfcs/0023-samhsa-v2.md:43]: Good catch, thanks!

**Review by MahiFentaye — APPROVED** [review]

**Review by mjburling — APPROVED** [review]

**Pull request merged**